### DISCUSSION TITLE
Feature request/question: Support for Shelly Plus Uni

### DISCUSSION THREAD

**Discussion by zdychacek** [mongoose-os-apps/shelly-homekit#1501] [Ideas]
Hi,

thank you for your work on this project. It's awesome, and I use your firmware daily in my light switches.

Now I am considering buying Shelly Plus Uni for controlling my entrance gate.

But I do not know if there is a support for Shelly Plus Uni (https://www.shelly.com/en-us/products/shop/shelly-plus-uni)? If not is it planned for near future?

Thank you for your response.

O.

**Comment by timoschilling** [comment]
I think a support with the same feature set as for the non plus uni will come.
But if you don't need a device which support 5V power supply, there is no need to use the plus version, as long as the non plus version is still on stock.

**Comment by nickmoone** [comment]
Any update on when support for the Shelly Plus Uni will be added? Isn't it just the same as the Shelly Uni with a different device ID?

**Comment by markirb** [comment]
No it is not. It is an ESP32 device instead of ESP8266. 

But still it should not be too hard.

Just needs time and hardware...

**Comment by uplyft** [comment]
I've got the hardware and I'm happy to help build this, if it wouldn't be too annoying for someone on the team to offer a bit of direction.

**Comment by markirb** [comment]
Just compare pull requests that add a new model (plus plug s for example). Then add pins you find on other projects like tasmota. If you have questions just feel free to ask

**Comment by rjmichna** [comment]
I'd like to use a Shelly Plus Uni as a Garage Door controller, and haven't been able to find the Non Plus version.  My thanks to anyone with the chops to code this!

**Comment by markirb** [comment]
A first version of the device is available in the release section as prerelease

**Comment by markirb** [comment]
Is added now

**Reply by uplyft** [reply]
Thank you! I upgraded the firmware to 1.4.4, then tried to install this version using the all devices url. Any way to diagnose why it might not have resulted in a successful install?

**Reply by markirb** [reply]
Can you try again? url mapping is sometimes a bit tricky

**Reply by uplyft** [reply]
Worked! Thanks so much!!

**Reply by uplyft** [reply]
@markirb Whoops... It worked the first time but now I'm getting the original behavior (simply not updating). Should I try the python script or learn how debugging OTA works? Thanks again!

**Reply by markirb** [reply]
Reboot. If it does not update, then enable logs and see what it says
